examples/mksomersaultmodifiedpdx.py

```diff
@@ -0,0 +1,230 @@
+#! /usr/bin/python3
+#
+# SPDX-License-Identifier: MIT
+#
+# This script demonstrates how an existing ODX database can be
+# modified programatically. Note that this is pretty hacky...
+import argparse
+from copy import deepcopy
+from typing import List, TypeVar
+
+import odxtools
+import odxtools.uds as uds
+from examples import somersaultecu
+from odxtools.diaglayer import DiagLayer
+from odxtools.diagservice import DiagService
+from odxtools.nameditemlist import NamedItemList
+from odxtools.odxlink import OdxLinkId, OdxLinkRef
+from odxtools.parameters.codedconstparameter import CodedConstParameter
+from odxtools.parameters.valueparameter import ValueParameter
+from odxtools.request import Request
+from odxtools.response import Response, ResponseType
+
+T = TypeVar("T")
+
+
+def find_named_object(item_list: List[T], name: str) -> T:
```

Comment:
if you really need such a function here, it means the current api is limited

Reply:
IMO this is pretty much a corner case that is only relevant when modifying database objects programmatically: If `item_list` is an instance of `NamedItemList`, this is basically equivalent to `item_list[name]`. The problem here is that the `.diag_comms` member of `DiagLayerRaw` is not a `NamedItemList` because it may also contain references (which do not exhibit a `.short_name` attribute)...